base/cvd/cuttlefish/host/libs/config/fetcher_config.cpp

```diff
@@ -68,7 +70,28 @@ std::ostream& operator<<(std::ostream& os, const CvdFile& cvd_file) {
   return os;
 }
 
+FetcherConfig::FetcherConfig() : mutex_(std::make_unique<std::mutex>()) {}
+
+FetcherConfig::FetcherConfig(FetcherConfig&& other) {
+  std::scoped_lock lock(*other.mutex_);
+
+  mutex_ = std::move(other.mutex_);
+  other.mutex_ = std::make_unique<std::mutex>();
+}
+
+FetcherConfig& FetcherConfig::operator=(FetcherConfig&& other) {
+  std::scoped_lock lock(*mutex_, *other.mutex_);
+
+  std::unique_ptr<std::mutex> to_be_deleted = std::move(mutex_);
+  mutex_ = std::move(other.mutex_);
```

Comment:
This isn't thread safe: Another thread could try to lock `other`'s mutex between this line and the next and end up dereferencing nullptr.

I can't think of a way to make the move constructor thread-safe with only the non-static class members.